### ISSUE TITLE
[lldb-dap] vscode - Core dump debugging not hitting breakpoints nor loading debug symbols?

### ISSUE THREAD

**Issue by borjamunozf** [llvm/llvm-project#110253]
Hello! I'm really a beginner with lldb-dap, looking for an alternatives and benchmarks regarding performance.

- Working config with vscode - gdb
```json
 {
      "name": "(gdb) Launch Core Dump",
      "type": "cppdbg",
      "request": "launch",
      "program": "/mnt/disk1/rcs/asd/bin/DispApp",
      "args": ["-g "],
      "stopAtEntry": false,
      "coreDumpPath": "${workspaceFolder}/coreDump",
      "sourceFileMap": {
          "/asdshell": "${workspaceFolder}",
      },
      "additionalSOLibSearchPath": "/mnt/disk1/asd/app/lib",
      "cwd": "${workspaceFolder}",
      "miDebuggerPath": "/usr/bin/gdb-multiarch",
      "MIMode": "gdb",
      "logging": {
        "engineLogging":"verbose",
        "trace": true,
        "traceResponse": true,
      },
}
```

- Trying to _mimick_ functionality with lldb-dap:
```json

 {
      "name": "(lldb-dap) Launch Core Dump",
      "type": "lldb-dap",
      "request": "attach",
      "coreFile": "${workspaceFolder}/coreDump",
      "program": "/mnt/disk1/asd/app/bin/DispApp",
      "sourceMap": [
        ["/asdshell", "${workspaceFolder}"]
        ],

      "initCommands":[
        "settings set target.source-map /asdshell ${workspaceFolder}",
        "settings set target.exec-search-paths /mnt/disk1/rcs/app/lib:${input:rcsosSysPath}/sysroots/x86-rcs-linux/",
      ],
      "internalConsoleOptions": "openOnFirstSessionStart"
      
    }
```

Info from breakpoints:
```
breakpoint list --verbose
(lldb) breakpoint list --verbose
Current breakpoints:
1: file = '/asdshell/asd/Main/Whatever/Source/WhateverInit.cpp, line = 13, exact_match = 0

2: file = '/asdshell/asd/Main/Whatever/Source/WhateverInit.cpp', line = 98, exact_match = 0
```

Show source map:
```
settings show target.source-map
(lldb) settings show target.source-map
target.source-map (path-map) =
[0] "/asdshell" -> "/home/borjamf/workspace/asd_shell"
```

![imagen](https://github.com/user-attachments/assets/18b2c208-8cde-419c-8e67-e05535c1540f)

I'll be happy to provide any info! Thanks a lot.

**Comment by borjamunozf** [comment]
Probaby related to #63973 

I guess there is no solution, we have to keep suffering cppdbg with gdb.

**Comment by vogelsgesang** [comment]
Seems to be a general lldb issue, not specific to lldb-dap

**Comment by jimingham** [comment]
There are two things reported here:

1) It looks like you don't have debug information for /asdshell/asd/Main/Whatever/Source/WhateverInit.cpp.  Does the symbol file you added actually have debug information for that file?
2) However, you do seem to have debug info for some other source file.  In that case, lldb was able to resolve the file & line spec to an address, and the "failure" was that it refused to write a trap instruction into the core file TEXT at that address.  That seems perfectly reasonable to me.  Why would it do any good to actually write trap instructions into a core file?

Actually, why is it important to set breakpoints when doing core file debugging at all?  Are you just using them to find addresses for functions or line specs?  If that's what you are trying to do, you should be fine.  The breakpoints that got resolved should show their addresses even if lldb didn't actually write a trap there.  But also, the `image lookup` command is a more straightforward way to do that task than pretending you want to set a breakpoint, so you can read out the resolved locations.  

The other minor warning was that there was something not quite right with the DWARF, and lldb warned you about that.  gdb used to have a `complaints` variable that would specify how many complaints to issue about odd looking debug info.  Back when I last used gdb, that was set to 0 by default.  Might be interesting to set that higher and see if gdb complains about this. Anyway, it's not uncommon to see these warnings, but that's more about the producer than the consumer.

**Comment by jimingham** [comment]
It would be useful here to separate possible problems with core files from problems with symbols.  To that end, can you run a process under lldb with the same executable and symbol files, and see if you see source information in places where with core files you weren't.  The speculation from the other report you linked to was that this was lack of support for some gcc DWARF constructs.  If this also happens just loading the executable & symbol file into lldb and running `source info -f some_source_file.c` should also fail.

**Comment by jimingham** [comment]
The title threw me off a bit because "why would you want to set breakpoints in a core file"...  And certainly you can't "hit" them since core files can't execute (at least on any system I know about).

**Comment by borjamunozf** [comment]
You're right about setting the breakpoints in the core dump, my fault. I mxied with other stuf I'm working on!
With the title I was referring for not reaching breakpoint it was more to _the core dump_ line or source code.

> It would be useful here to separate possible problems with core files from problems with symbols. To that end, can you run a process under lldb with the same executable and symbol files, and see if you see source information in places where with core files you weren't. The speculation from the other report you linked to was that this was lack of support for some gcc DWARF constructs. If that's what is going on, just loading the executable & symbol file into lldb and running `source info -f some_source_file.c` should also fail.

So my issue, as whole, its:
- the core dump is not hitting or reaching anywhere as the GDB config one. I think the sourceMap it's not working properly, but I'll get more info asap.

I'll check what you said and come back to you
